### PR TITLE
Added callback progression

### PR DESCRIPTION
**Pull request by crup**
- Added callback for progression, it returns current value and end value in callback function. Defaults to negative progression. 
- Upgraded React deps in example folder and migrated to React hooks example

### CODE REVIEW

**Comment by crup** [comment]
@Bryze Maybe we could come up with better api or maybe better prop names in next few iterations.

File: src/index.js [hunk 1]
```diff
@@ -10,23 +10,19 @@ const ReactTimer = ({
...
   const [value, setValue] = useState(start)
 
-  const computeResult = () => progression === DESCENDING ? (value - delta) : (value + delta)
-
-  const evaluateCondition = () => progression === DESCENDING ? (value <= end) : (value >= end)
-
   useEffect(() => {
     const timer = window.setInterval(() => {
-      if (evaluateCondition()) {
+      if (value === end) {
```
Comment (Bryze) [inline src/index.js]: If we are abstracting the implementation of the `progression`, then maybe, we should abstract this expression into a prop that returns a boolean value?

File: example/src/App.js [hunk 2]
```diff
@@ -1,21 +1,29 @@
...
+        delta={2}
+        onComplete={(value) => console.log('ENDED WITH VALUE', value)}
+        onTick={(value) => console.log('CURRENT VALUE', value)}
+      >
+        {(time) => <div>{time}</div>}
+      </ReactTimer>
+      <ReactTimer
+        start={0}
+        end={30}
+        delta={2}
```
Comment (Bryze) [inline example/src/App.js]: I think the prop `delta` pretty much becomes unused since we have abstracted the progression into a user-defined callback

**Comment by Bryze** [comment]
declining this pr as the proposed changes don't fit with the new roadmap in mind

**Pull request closed without merging**